### ISSUE TITLE
Weighted_point_3 does not name a type

### ISSUE THREAD

**Issue by DahnJ** [CGAL/cgal#3171]
## Issue Details

I installed Linux Mint 18.3 MATE, then g++ and libcgal-dev through sudo apt-get. However, parts of the library seem to not be accessible. For example, using the class Point_3 works, whereas Weighted_point_3 does not:

`regular2.cpp:6:12: error: ‘Weighted_point_3’ in ‘K {aka class CGAL::Epick}’ does not name a type
 typedef K::Weighted_point_3                            Weighted_point;
`

## Source Code
Minimum code to reproduce the error above on my machine:

``` c++
#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
#include <CGAL/Regular_triangulation_3.h>
typedef CGAL::Exact_predicates_inexact_constructions_kernel K;
typedef K::Weighted_point_3                            Weighted_point;
int main()
{
  return 0;
}

```

## Environment

* Operating system Linux Mint 18.3 MATE:
* Compiler: g++ 5.4.0.
* CGAL version: 4.7-4

## Additional details

- [Example 5.1](https://doc.cgal.org/latest/Triangulation_3/index.html#title10) from 3d triangulations works
- [Example 5.5](https://doc.cgal.org/latest/Triangulation_3/index.html#title19) from 3d triangulations gives `error: ‘get_default_random’ is not a member of ‘CGAL’`

**Comment by sloriot** [comment]
The doc you are looking at is that of CGAL 4.12, if you can't upgrade look at https://doc.cgal.org/4.7/Triangulation_3/index.html#title10

**Comment by lrineau** [comment]
`Weighted_point_3` was added in the `Kernel` concept only in CGAL-4.10. And the function `get_default_random()` as well (added only in CGAL-4.9).

Have a look at the same [example 5.5 from the documentation of CGAL-4.7](https://doc.cgal.org/4.7/Triangulation_3/index.html#title19).

**Comment by DahnJ** [comment]
Thanks all. Is there a way for me to install CGAL 4.12? I tried following the [FAQ entry](https://www.cgal.org/FAQ.html#debian_packages), i.e. I've appended sources.list and installed apt-transport-https, but `sudo apt-get install libcgal-dev` still installs version 4.7.  Am I misunderstanding the process? I am entirely new to linux, so I am prone to misunderstand many details an experienced user wouldn't.
